app/Http/Controllers/Api/Web/StatsController.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * Pony.fm - A community for pony fan music.
+ * Copyright (C) 2015 Peter Deltchev
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Poniverse\Ponyfm\Http\Controllers\Api\Web;
+
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Poniverse\Ponyfm\Http\Controllers\ApiControllerBase;
+use Poniverse\Ponyfm\Models\Track;
+use Auth;
+use DB;
+use Response;
+use Carbon\Carbon;
+
+class StatsController extends ApiControllerBase
+{
+    public function getTrackStatsHourly($id)
+    {
+        $query = DB::table('resource_log_items')
+            ->selectRaw('created_at AS time, COUNT(1) AS `plays`')
+            ->where('track_id', '=', $id)
+            ->where('log_type', '=', 3)
+            ->whereRaw('`created_at` > now() - INTERVAL 1 DAY')
+            ->groupBy('created_at')
```

Comment:
Unless I'm missing something, the utility of this `group by` statement is limited as it will only group together plays that happened within the exact same second. Was that the intent?